client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,146 @@
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		const defaultWidth = 640;
+		const defaultAspectRatio = 16 / 9;
+		let { width, height } = this.props.video
+			? this.props.video
+			: {
+				width: defaultWidth,
+				height: defaultWidth / defaultAspectRatio
+			};
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			const definedAspectRatio = shortcodeWidthAttribute / shortcodeHeightAttribute;
+			if ( definedAspectRatio > aspectRatio ) {
+				width = shortcodeHeightAttribute * aspectRatio;
+				height = shortcodeHeightAttribute;
+			} else {
+				width = shortcodeWidthAttribute;
+				height = shortcodeWidthAttribute / aspectRatio;
+			}
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions(
+			parseInt( namedAttrs.w, 10 ),
+			parseInt( namedAttrs.h, 10 )
+		);
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: namedAttrs.autoplay === 'true',
+			hd: namedAttrs.hd === 'true',
+			loop: namedAttrs.loop === 'true',
+			at: parseInt( namedAttrs.at, 10 ) || 0,
+			defaultLangCode: namedAttrs.defaultlangcode
+		};
+
+		return Object.assign(
+			{},
+			pick( shortcodeAttributes, ['guid', 'w', 'h'] ),
+			{ embedUrl: this.getEmbedUrl( shortcodeAttributes ) }
+		);
+	}
+
+	getEmbedUrl( shortcodeAttributes ) {
+		const defaultAttributeValues = { hd: false, at: 0, defaultLangCode: undefined };
+		const attributesWithNonDefaultValues = omitBy(
+			shortcodeAttributes,
+			( value, key ) => defaultAttributeValues[key] === value );
+		const queryStringAttributes = ['autoplay', 'hd', 'loop', 'at', 'defaultLangCode'];
+		const queryString = QueryString.stringify( pick( attributesWithNonDefaultValues, queryStringAttributes ) );
+
+		return `https://videopress.com/embed/${ shortcodeAttributes.guid }?${ queryString }`;
+	}
+
+	onLoad() {
+		const doc = this.props.editor.iframeElement.contentDocument;
+		const script = doc.createElement( 'script' );
+		script.src = 'https://videopress.com/videopress-iframe.js';
```

Comment:
>I would keep the existing script so that we do not add knowledge to Calypso about how VideoPress videos are rendered: we provide an iframe, the rest is taken care if by VideoPress.

Okay, makes sense :+1: